### DISCUSSION TITLE
Add a new board

### DISCUSSION THREAD

**Discussion by nicola-lunghi** [Builty/TonexOneController#42] [Q&A]
Hi I have two ESP32-S3 board here
This is the model I have [LiliGo T-DISPLAY-S3 CAP](https://it.aliexpress.com/item/1005005416973021.html?src=google&pdp_npi=4%40dis%21EUR%2114.69%2114.69%21%21%21%21%21%40%2112000037269587585%21ppc%21%21%21&src=google&albch=shopping&acnt=272-267-0231&isdl=y&slnk=&plac=&mtctp=&albbt=Google_7_shopping&aff_platform=google&aff_short_key=UneMJZVf&gclsrc=aw.ds&&albagn=888888&&ds_e_adid=&ds_e_matchtype=&ds_e_device=c&ds_e_network=x&ds_e_product_group_id=&ds_e_product_id=it1005005416973021&ds_e_product_merchant_id=109240285&ds_e_product_country=IT&ds_e_product_language=it&ds_e_product_channel=online&ds_e_product_store_id=&ds_url_v=2&albcp=20730495829&albag=&isSmbAutoCall=false&needSmbHouyi=false&gad_source=1&gclid=Cj0KCQiAsaS7BhDPARIsAAX5cSA4ZfJ0KyiYDSMMAGUawiew2qBxYln2Zl595PJh1ffN4nUecVWvjysaAl7XEALw_wcB)
How can I add it to the repo?
also can I contribute to a bit of CI or there's a reason that you are not putting it in?

**Comment by Builty** [comment]
Hi Nicola.
I have updated the Firmware Development doc here to add the steps needed to add a new hardware platfform.
It's quite east, apart from the display part. It will need a new UI design done, however I can help out there if needed.

Lilygo do have an ESP-IDF display driver sample, which is helpful: https://github.com/Xinyuan-LilyGO/LilyGo-Display-IDF/blob/master/main/display_s3.c

However, first step is to make sure the hardware can do USB OTG. Looking at the schematic:
- USB-C port connects to VBUS
- GPIO header has connection for +5V
- Diode allows power from VBUS to get to +5V, but blocks the reverse direction (but maybe I missed something)

For the Tonex to be powered from the board, we need to connect 5v to somewhere and have it come out of VBUS.
I would suggest doing some tests on this board before doing any coding, to make sure its possible to achieve this.

![image](https://github.com/user-attachments/assets/bae8cdab-0491-4267-b6d6-0d89a6da2dc4)
